datamade/how-to
# Automatically update cookiecutter dependencies

## Overview

This branch adds a github action to automatically update cookiecutter dependencies from pypi and node. There are dependency files here that are already updated, and you can see the process that did that in the Actions tab. We're using `pip-upgrader` and `npm-check-updates` to update the dependency files, and then committing/pushing the result.

At time of writing it's scheduled to run every Sunday at 5:30 UTC, but I can definitely change this.

The only file that hasn't been updated is the `requirements.txt` in the `new-wagtail-app` directory. This was intentional so we could have a clearly outdated file to test with.

- Handles #354

### Notes
After this pr is approved, I'll just have to change the branch the action is pushing to, to `main`.

I was able to test the update locally and here when this branch got a push event. Evidence of that is in the workflow in the Actions tab and also the commit here titled "Update pypi and node dependencies". However, [the github docs on scheduled jobs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) say that cron tasks only work on the default branch, so I don't think we'll be able to test that until this gets merged.

## Testing Instructions

* Checkout this branch and add a push event to main.yml:
```
on:
  schedule:
    - cron: '30 5 * * 0'
  push:
    branches:
      - feature/dependency-auto-update
```
* Commit and push to this branch
  * Confirm that the wagtail app's `requirements.txt` updated on github
* Remove the push trigger and push back up
